src/abc/io.jl

```diff
@@ -145,6 +146,14 @@ struct MeanVarEmulatedParticleSelection <: AbstractEmulatedParticleSelection
 end
 MeanVarEmulatedParticleSelection() = MeanVarEmulatedParticleSelection(1.0)
 
+"""
+   PosteriorSampledEmulatedParticleSelection <: AbstractEmulatedParticleSelection
+
+When this strategy is used, the distance is sampled from the GP posterior of the [`gp_regression`](@ref)
+object. If the sampled distance is below the threshold the particle is accepted.
+"""
+struct PosteriorSampledEmulatedParticleSelection <: AbstractEmulatedParticleSelection end
```

Comment:
Should we give the user control whether the full covariance matrix is used or not? It is `true` by default, which can be rather expensive without giving much accuracy improvement, compared to using just the diagonal. Suggest we add the `use_full_covariance` flag as a member of this struct, make it a parameter in `gp_regression_sample`, and pass it on from `abc_select_emulated_particles`. And the default value should be `false`, IMHO.

Reply:
I agree we should add a field to the struct, but I will do some tests to check the effect on the computation time.

If possible we should use the full covariance as sampling from a diagonal-covariance approximation of the GP posterior is very different to sampling from the exact posterior. The extent to which this difference affects the eventual ABC posterior will need to be checked (I will do this and add it to `abc-example.ipynb`) but my feeling is that it could have quite a large effect.